go/ql/test/experimental/CWE-176/example_bad.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"html"
+	"net/http"
+	"strings"
+
+	"golang.org/x/text/unicode/norm"
+)
+
+func main() {}
+
+func bad() {
+	http.HandleFunc("/bad1", func(w http.ResponseWriter, req *http.Request) {
+
+		unicode_input := req.URL.Query().Get("unicode_input")
+		escaped := html.EscapeString(unicode_input)
+		unicode_norm := norm.NFKC.String(escaped) // $result=BAD
+		fmt.Println(w, "Results: %q", unicode_norm)
+	})
+
+	http.HandleFunc("/bad2", func(w http.ResponseWriter, req *http.Request) {
+
+		unicode_input := req.URL.Query().Get("unicode_input")
+		if strings.IndexAny(unicode_input, "<﹤＜") == -1 {
+			unicode_norm := norm.NFKC.String(unicode_input) // $result=BAD
+			fmt.Println(w, "Results: %q", unicode_norm)
+		}
+
+	})
+
+	http.HandleFunc("/bad3", func(w http.ResponseWriter, req *http.Request) {
+
+		unicode_input := req.URL.Query().Get("unicode_input")
```

Comment:
To check this is working as you hope, I suggest adding some cases that should not be flagged, like norming something in the `else` block of that `ContainsAny` statement, or just norming something with no validation operations involved.

Reply:
I will be working on this, this weekend sorry about the delay...

Reply:
I have rewritten the test cases according to the commit [3ad7088](https://github.com/github/codeql/pull/12994/commits/3ad70880c0173164690bb82b42494f54829d0e70).